Feature: Mars Climate Database Atmosphere models

valeriof7 opened the pull request:
PR for issue #382

valeriof7 committed:
Add docs for observation ancillary settings

valeriof7 committed:
Add docs for observation dependent variables

valeriof7 committed:
Add docs for observation wrappers

valeriof7 committed:
Add docs observation biases

DominicDirkx committed:
Merge pull request #498 from valeriof7/doc/observationCollection

[Doc]: Add docs for the observation module

github-actions[bot] committed:
Apply clang-format

larshinueber committed:
add Python class constructors to docs, enable inheritance hints by default

larshinueber committed:
make parameter ordering warning optional

actions-user committed:
Bump version: 1.0.0.dev11 → 1.0.0.dev12

larshinueber committed:
Merge branch 'feature/optional-parameter-warning' into feature/horizons-constructor-docs

larshinueber committed:
expose apriori and consider covariances

luigigisolfi committed:
Added 80col file parser and improved mpc.py

luigigisolfi committed:
Added 80col file parser and improved mpc.py

luigigisolfi committed:
added files

luigigisolfi committed:
Merge branch 'develop' of https://github.com/tudat-team/tudatpy into feature/80col_parser

luigigisolfi committed:
Added 80col file parser and improved mpc.py

luigigisolfi committed:
Merge branch 'feature/80col_parser' of https://github.com/luigigisolfi/tudatpy into feature/80col_parser

luigigisolfi committed:
Improved and updated Horizons Wrapper ephemerides() function

luigigisolfi committed:
Corrected typo in function names

luigigisolfi committed:
Corrected return for parse_80cols_file (returns astropy table)

DominicDirkx committed:
Merge pull request #502 from larshinueber/feature/horizons-constructor-docs

Add Horizons Constructor docs, make parameter order warning optional

github-actions[bot] committed:
Apply clang-format

DominicDirkx committed:
Working on refactoring

jo11he committed:
eliminate c++17 structured bindings

jo11he committed:
purge last structured decomposition

actions-user committed:
Bump version: 1.0.0.dev12 → 1.0.0.dev13

luigigisolfi committed:
Added 80cols parser unit tests

luigigisolfi committed:
Added 80cols parser unit tests

DominicDirkx committed:
Corrected unit test

DominicDirkx committed:
Correctec compile error

DominicDirkx committed:
Re-enabling type checking

DominicDirkx committed:
Corrected exposure

DominicDirkx committed:
Merge pull request #508 from tudat-team/fix/506

Resolved #506

github-actions[bot] committed:
Apply clang-format

luigigisolfi committed:
Merge branch 'develop' of https://github.com/tudat-team/tudatpy into feature/80col_parser

luigigisolfi committed:
Final version of 80col_parser, corrected time scale conversion for historical dates and allow for utc-tai conversion

actions-user committed:
Bump version: 1.0.0.dev13 → 1.0.0.dev14

luigigisolfi committed:
Added small improvement to horizons_wrapper.py

actions-user committed:
Bump version: 1.0.0.dev14 → 1.0.0.dev15

DominicDirkx committed:
Correct partial

jo11he committed:
Merge branch 'feature/interface_for_mean_tidal_potential_100' of github.com:jo11he/my-tudatpy into feature/interface_for_mean_tidal_potential_100

jo11he committed:
improve default handling mean tidal forcing

jo11he committed:
correct function signature in header

jo11he committed:
eliminate constructor mismatch from parameter unit test

actions-user committed:
Bump version: 1.0.0.dev15 → 1.0.0.dev16

luigigisolfi committed:
small edits

luigigisolfi committed:
small edits to mpc.py

luigigisolfi committed:
Fixed typo: note1 and note2 retrieval

DominicDirkx committed:
Merge small variable naming conventions corrections

DominicDirkx committed:
Merge pull request #434 from jo11he/feature/interface_for_mean_tidal_potential_100

Feature/interface for mean tidal potential 100

github-actions[bot] committed:
Apply clang-format

DominicDirkx committed:
Merged

DominicDirkx committed:
Corrected merge

DominicDirkx committed:
Corrected signatures

actions-user committed:
Bump version: 1.0.0.dev16 → 1.0.0.dev17

luigigisolfi committed:
Added handling of minimum representable epoch on linux systems

luigigisolfi committed:
changed boost check in unitTestDateTime.cpp

DominicDirkx committed:
Corrected Doppler partials

luigigisolfi committed:
Reverted changes to dateTime.h from commit  f2112edcaf1c806cb6bf0af714e515210b4d3e62

luigigisolfi committed:
ctest and pytest all good

DominicDirkx committed:
Merge pull request #511 from tudat-team/fix/510

Correct Doppler IFMS partials

github-actions[bot] committed:
Apply clang-format

actions-user committed:
Bump version: 1.0.0.dev17 → 1.0.0.dev18

DominicDirkx committed:
Added improved multi-arc support

DominicDirkx committed:
Update

luigigisolfi committed:
Fix relative path, add eros_obs.txt file to support unit test, small edits in mpc.py imports

DominicDirkx committed:
Merge pull request #489 from jo11he/feature/rtg_acceleration_parameter_across_multi_arcs

Feature/rtg acceleration parameter across multi arcs

github-actions[bot] committed:
Apply clang-format

DominicDirkx committed:
Corrections to new multi-arc ephemeris

luigigisolfi committed:
Resolved few suggestions from PR #505

actions-user committed:
Bump version: 1.0.0.dev18 → 1.0.0.dev19

DominicDirkx committed:
Update multi-arc ephemeris

luigigisolfi committed:
small edits

DominicDirkx committed:
Working on new multi-arc test

DominicDirkx committed:
Corrected test

DominicDirkx committed:
Completed multi-arc test

DominicDirkx committed:
Merged

DominicDirkx committed:
COrrectec

DominicDirkx committed:
Corrected valid interpolation

luigigisolfi committed:
Merge branch 'develop' of https://github.com/tudat-team/tudatpy into feature/80col_parser

DominicDirkx committed:
Added multi-arc docstrings

DominicDirkx committed:
Added to .rst

DominicDirkx committed:
Merge pull request #512 from tudat-team/fix/492

Added improved multi-arc support

github-actions[bot] committed:
Apply clang-format

luigigisolfi committed:
Addressed all comments on PR #505. Added parser_80col submodule. Wrote coherent docstrings. Added to_roman() support function in tudatpy.utils._support. Avoided duplicated functionalities within mpc.py. PR is now ready.

luigigisolfi committed:
Slightly modified the unit test to make it more explicit (and not to fail). Small edits to get_observations function (uniformed astroquery designation to 80cols parser designation).

actions-user committed:
Bump version: 1.0.0.dev19 → 1.0.0.dev20

DominicDirkx committed:
Modied docstrings

DominicDirkx committed:
Added (not fully working) differenced time of arrival observation model and aprtials

DominicDirkx committed:
Corrected test

actions-user committed:
Bump version: 1.0.0.dev20 → 1.0.0.dev21

DominicDirkx committed:
Modified observation model to allow UTC time difference

DominicDirkx committed:
Expanded unit test

actions-user committed:
Bump version: 1.0.0.dev21 → 1.0.0.dev22

DominicDirkx committed:
Added exposure

DominicDirkx committed:
Corrected partial; added test

DominicDirkx committed:
Modifying parameter order

DominicDirkx committed:
Attempting fix of consider parameters

DominicDirkx committed:
Corrected consider parameter bookkeeping

actions-user committed:
Bump version: 1.0.0.dev22 → 1.0.0.dev23

DominicDirkx committed:
Merge pull request #515 from tudat-team/feature/differenced_time_frequency

Differenced time of arrival observation model and partials

DominicDirkx committed:
Merge pull request #519 from tudat-team/feature/torque_partials_fix

Corrected spherical harmonic partial; added missing test

github-actions[bot] committed:
Apply clang-format

luigigisolfi committed:
Slightly modified the unit test to make it more explicit (and not to fail). Small edits to get_observations function (uniformed astroquery designation to 80cols parser designation).

DominicDirkx committed:
Corrected use of consider parameters

actions-user committed:
Bump version: 1.0.0.dev23 → 1.0.0.dev24

jo11he committed:
add arcwise version of parameters to enum, utils related to enum (.cpp)

jo11he committed:
create new arc-wise parameter class

jo11he committed:
make additional functions in acceleration model that is required to interface with arc-wise parameter

DominicDirkx committed:
Corrected consider parameter book-keeping; covariance propagation to be done

valeriof7 committed:
Fix "position" typo

jo11he committed:
correct double parameter bool of new arcwise parameter

jo11he committed:
add arcwise parameyer settings class and factory functions for settings objects

jo11he committed:
add arcwise parameter case to acceleration model checks

jo11he committed:
add case for new arcwise parameter construction in big createVectorParameterToEstimate factory function

jo11he committed:
implement partials for arcwise scaling parameters / refactor AerodynamicsPartial class for consistency

valeriof7 committed:
Fix API of astro module

valeriof7 committed:
Fix data module

valeriof7 committed:
Fix environment_setup/atmosphere

valeriof7 committed:
Fix constants module

DominicDirkx committed:
Added test; docstrings

jo11he committed:
fix syntax for successful build (incl. passing unit test of refactored aerodynamics partials)

DominicDirkx committed:
Added propagation of consider covariance

DominicDirkx committed:
Modification to docs

jo11he committed:
unit test implementation and trouble shooting

DominicDirkx committed:
Final corrections

DominicDirkx committed:
Merge pull request #522 from valeriof7/fix/521

Fix "position" typo

DominicDirkx committed:
Merged

DominicDirkx committed:
Merge pull request #520 from tudat-team/feature/consider_parameter_fix

Feature/consider parameter fix

github-actions[bot] committed:
Apply clang-format

actions-user committed:
Bump version: 1.0.0.dev24 → 1.0.0.dev25

luigigisolfi committed:
removed type:ignore comments from mpc.py

luigigisolfi committed:
move to_roman() to transform_integer_to_roman_number()

luigigisolfi committed:
added type hints to dicts

luigigisolfi committed:
removed wrong last line from unit test

jo11he committed:
add unit test arcwise scaling partials

DominicDirkx committed:
Added checks for single-arc processing settings

jo11he committed:
fix data location and parameter setup bug for unit test

jo11he committed:
implement fix by adding fabs to second member of comparison statement

DominicDirkx committed:
Merge pull request #528 from jo11he/fix/multi_arc_parameter_setup_bug

w.r.t. #527: implement fix by adding fabs to second member of comparison statement

DominicDirkx committed:
Added reset workaround from multi-arc

actions-user committed:
Bump version: 1.0.0.dev25 → 1.0.0.dev26

DominicDirkx committed:
Merge pull request #526 from tudat-team/fix/474

Added checks for single-arc processing settings

github-actions[bot] committed:
Apply clang-format

jo11he committed:
fix copy paste error, cause runtime error

jo11he committed:
add overload to simulatePseudoObservations for easier multi-arc use

actions-user committed:
Bump version: 1.0.0.dev26 → 1.0.0.dev27

valeriof7 committed:
mcd compilation

valeriof7 committed:
Remove conda environments checks in the cmake files and rely on the arguments of the cmake call in `builds.py` and `cmakepresets.json`

valeriof7 committed:
Fix conditional building

valeriof7 committed:
Fix and enhance compilation

valeriof7 committed:
Tested atmospheric model

DominicDirkx committed:
Merged

jo11he committed:
fix wrong enum in settings factory function

jo11he committed:
add unit test for estimation of arcwise aero scaling (+ added arcwise constant drag)

jo11he committed:
remove cout statements from unit test

jo11he committed:
expose enum and settings factory functions for new arcwise aerodynamic scaling parameters

DominicDirkx committed:
Merge branch 'develop' into feature/arcwise_aerodynamic_scaling_parameters

DominicDirkx committed:
Corrected exposure

DominicDirkx committed:
Corrects #535

DominicDirkx committed:
Update observationCollection.h

DominicDirkx committed:
Corrected

DominicDirkx committed:
Merge pull request #538 from tudat-team/fix/474b

Fix/474b

DominicDirkx committed:
Merge pull request #537 from tudat-team/fix/535

Fix/535

github-actions[bot] committed:
Apply clang-format

DominicDirkx committed:
Merge pull request #525 from tudat-team/feature/docstrings_module_updates

Modied docstrings

github-actions[bot] committed:
Apply clang-format

DominicDirkx committed:
Merge pull request #505 from luigigisolfi/feature/80col_parser

Feature/80col parser (closes sub-issues #504 & #505)

github-actions[bot] committed:
Apply clang-format

DominicDirkx committed:
Added missing function calls

alfonsoSR committed:
Update environment_setup init

This PR (https://github.com/tudat-team/tudatpy/pull/525) removed some
deprecated functions from the `environment_setup` submodule of the
kernel, but they were still listed in `environment_setup/__init__.py`.
This would cause an error when trying to import the module.

This commit fixes the issue by removing `set_aerodynamic_guidance`,
`set_aerodynamic_orientation_functions`, and
`set_constant_aerodynamic_orientation` from
`environment_setup/__init__.py`.

jo11he committed:
add overload cast in exposure to newly overloaded function

DominicDirkx committed:
Merge pull request #541 from tudat-team/fix/deprecated_functions_environment_setup

Remove deprecated functions from `environment_setup/__init__.py`

DominicDirkx committed:
Merge pull request #540 from tudat-team/fix/514

Added missing function calls

actions-user committed:
Bump version: 1.0.0.dev27 → 1.0.0.dev28

DominicDirkx committed:
Merge pull request #536 from jo11he/feature/arcwise_aerodynamic_scaling_parameters

Feature/arcwise aerodynamic scaling parameters

github-actions[bot] committed:
Apply clang-format

DominicDirkx committed:
Corrected test

actions-user committed:
Bump version: 1.0.0.dev28 → 1.0.0.dev29

DominicDirkx committed:
Update test_dependent_variable_dictionary.py

actions-user committed:
Bump version: 1.0.0.dev29 → 1.0.0.dev30

DominicDirkx committed:
Re-enabled test

DominicDirkx committed:
Added docstrings and exposure

DominicDirkx committed:
Merge pull request #546 from tudat-team/fix/542

Added docstrings and exposure

github-actions[bot] committed:
Apply clang-format

DominicDirkx committed:
Corrected test path for Azure consistency

DominicDirkx committed:
Merged

DominicDirkx committed:
Merged

DominicDirkx committed:
Enhance description of high-precision constants usage

Clarify the conditions under which high-precision constants are beneficial, specifically noting the reliance on C++ and compiler architecture for achieving 80-bit precision.

DominicDirkx committed:
Modified integrator and propagator settings setup description (including termination settings)

DominicDirkx committed:
Modified docs; corrected compiler error

DominicDirkx committed:
Merged

DominicDirkx committed:
Corrected iso time string generation

DominicDirkx committed:
Hybrid-arc dependent variable interface corrected

DominicDirkx committed:
Document Time class in time_representation

Added documentation for the Time class in time_representation.rst.

DominicDirkx committed:
Merge pull request #549 from tudat-team/fix/481

Hybrid-arc dependent variable interface corrected

DominicDirkx committed:
Merge pull request #548 from tudat-team/fix/463

Corrected iso time string generation

github-actions[bot] committed:
Apply clang-format

actions-user committed:
Bump version: 1.0.0.dev30 → 1.0.0.dev31

DominicDirkx committed:
Modified estimation docstrings

DominicDirkx committed:
Docs update

DominicDirkx committed:
Expanded docs for ancillary settings and observtion models

DominicDirkx committed:
Merge pull request #547 from tudat-team/feature/api_docs_enhancement

Expansion of module-level docstrings

github-actions[bot] committed:
Apply clang-format

actions-user committed:
Bump version: 1.0.0.dev31 → 1.0.0.dev32

actions-user committed:
Bump version: 1.0.0.dev32 → 1.0.0.dev33

DominicDirkx committed:
Increase tolerance for Phobos and Mars position checks

Updated tolerance for position checks in torque partials tests.

jo11he committed:
add new rotation_pole_libration_terms parameter type to enums

DominicDirkx committed:
Updates to exposure and docstrings

jo11he committed:
add parameter class for pole librations and make interface with IAU rotation model class

DominicDirkx committed:
Updated examples submodule

DominicDirkx committed:
Merged

jo11he committed:
add parameter settings class and case in createVectorParameterToEstimate( )

jo11he committed:
change parameter size to flat vector and adjust interface to rotation model class accordingly

DominicDirkx committed:
Modified docs

jo11he committed:
add rotation matrix partials w.r.t. new parameters

jo11he committed:
add new parameter case for createRotationMatrixPartialsWrtParameter

actions-user committed:
Bump version: 1.0.0.dev33 → 1.0.0.dev34

jo11he committed:
fix implementation errors for successful build

DominicDirkx committed:
Merge pull request #554 from tudat-team/feature/api_docs_enhancement

Modified estimation module-level docstrings

github-actions[bot] committed:
Apply clang-format

actions-user committed:
Bump version: 1.0.0.dev34 → 1.0.0.dev35

actions-user committed:
Bump version: 1.0.0.dev35 → 1.0.0.dev36

jo11he committed:
add pole terms to unit test and correction to partials formulation

jo11he committed:
expose iau_rotation_model_pole_librations parameter factory function and complete set of iau parameter types

DominicDirkx committed:
Merged

DominicDirkx committed:
Merged

jo11he committed:
make parameter names in python interface consistent

jo11he committed:
correct get by reference functions

DominicDirkx committed:
Refactor ancillary settings property to readonly

github-actions[bot] committed:
Apply clang-format

DominicDirkx committed:
Modified API docs

DominicDirkx committed:
Merge pull request #557 from tudat-team/feature/api_docs_enhancement

Feature/api docs enhancement

actions-user committed:
Bump version: 1.0.0.dev36 → 1.0.0.dev37

DominicDirkx committed:
Merge pull request #555 from jo11he/feature/iau_pole_libration_parameters

Feature/iau pole libration parameters

github-actions[bot] committed:
Apply clang-format

actions-user committed:
Bump version: 1.0.0.dev37 → 1.0.0.dev38

valeriof7 committed:
Use altitude from flightConditions

valeriof7 committed:
Merge remote-tracking branch 'upstream/develop' into mcd

valeriof7 committed:
Update pybind exposure

valeriof7 committed:
Add MCData and source code to gitignore

valeriof7 committed:
Add fortran module files in gitignore

valeriof7 committed:
Fix gitignore

valeriof7 committed:
Merge branch 'mcd' into feature/mcd

DominicDirkx commented:
Great to add this! A couple of questions:
* Are any additional package dependencies needed except for mcd? I see netcdf, fortran compilers, and maybe other things(?)
* Why does this PR  say it consists of a total of 6306 commits? The changed files seem fine, so seems to be some git-related messiness
* Any idea how to test this in our CI? We could put mcd in a private repo, and give the test access to the private repo to clone it as part of the test (?) Is the implementation platform-independent?

valeriof7 commented:
In order:
- I have summarized everything in issue #382 . In brief, the dependencies are Fortran and NetCDF-Fortran.
- Yes, that's exactly what happened. I committed the source code and data of MCD; therefore, I used Git filter-repo to remove the files from the commits. I think they were somehow modified in the hash. I'd need help in case this needs to be fixed.
- Not really, to be honest, because many steps have to be performed manually. The dependencies listed above are platform dependent, and the MCD source code and data have to be requested by each user. As far as my cmake knowledge goes, the implementation should be platform independent.

valeriof7 commented:
I know for sure on linux/macOs, can't confirm on Windows

valeriof7 committed:
Merge remote-tracking branch 'upstream/develop' into feature/mcd

DominicDirkx committed:
Update develop

DominicDirkx committed:
Update to latest develop

DominicDirkx commented:
(updated branch to latest develop)

valeriof7 commented:
I know that Nicoló is working on a CI workflow to install the necessary dependencies. Should we wait to test issue #419?

DominicDirkx commented:
You can go ahead and test 419 with this branch, if it's succesful in the current setup we can merge it. The automated testing will be a separate point to handle

valeriof7 committed:
Optimize McdAtmosphereModel.

- Caching layer to avoid redundant MCD calls: cachedAltitude_/Longitude_/Latitude_/Time_, physically-motivated tolerances (time, horizontal, vertical), and cache invalidation logic.
- Optional extra-variable computation: computePropertiesIfNeeded supports on-demand extvar(14-85) retrieval and tracks lastComputedWithExtras_.
- Exposes getters for always computed and auxiliary extvars: density, pressure, temperature, zonal/meridional winds, mean/extravariables, radial distance, areoid/surface altitudes, orography, slope, Ls, LTST, solar angles, etc.
- Speed-of-sound calculation using MCD-provided gamma and R (with Mars defaults fallback).

nikemais commented:
Since more models in the future will rely on MCD (for example, PRP),  I think MCD should have its own class and updater, which gets called by the  EnvironmentUpdater. This way, only one call of MCD (with the correct variable keys set during setup) has to be made at each step, and then the models can share and access the requested variables

DominicDirkx commented:
Agreed, to be added in a next PR (?)

nikemais commented:
Yes, we can merge this. I will create the issue for the next PR with the list of features

valeriof7 commented:
Before the PR, someone else needs to take a detailed look at the implemented caching logic and the unit tests. The former is based on Section 5.5 of the [UserGuide](https://www-mars.lmd.jussieu.fr/mars/info_web/user_manual_6.1.pdf), and the latter is based on the ten unit tests provided in the MCD software directory `MCD_6.1/mcd/testcase`.

Regarding the last comment, as I mentioned to Nicoló offline, the new implementation of the call should be easy to transfer to a dedicated class that wraps the call to `mcd.f90`.

nikemais commented:
I will have a look at the caching logic and unit tests (especially to understand the minimal amount of data they can run on)

valeriof7 committed:
Fix bug for arc-wise drag, side, and lift component scaling factors in Cartesian state partials

nikemais commented:
I had some time to review the caching logic, and everything looks good. 

I'd like to express a small concern regarding the input altitude. In Tudat, the computation depends on the type of shape model chosen for Mars, which defaults to a spherical model. By setting MCD's zkey=3 (altitude above local surface, computed with precise models) and by not performing a conversion on the input altitude (for example, to an oblate-spherical altitude), it can lead to noticeable differences. To solve this, I suggest the use of zkey=1 (radial distance from the centre).

Aside from this, I think this could be merged! The next PR will focus on CI testing for MCD, followed by restructuring.

DominicDirkx commented:
Great, and agreed on the default setting

valeriof7 commented:
I think I chose this option for two reasons:
- First, if you set the distance from the center, you lose the ability to model the body's shape as you wish. What do you mean by "not performing a conversion"? As far as I know, the flight condition object knows which shape model to use for geodetic altitude computation, so everything should be consistent.
- Second, to replicate the provided test cases, which use zkey=3. If we change that, we need to remember to update the unit tests as well.

DominicDirkx commented:
Having radial distance to center of Mars is more robust, I think. Presumably MCD has an internal model for the Martian shape (since it can work with distance to center of Mars). Any shape model we use will be inconsistent with this model (even if only slightly). To make sure the atmosphere model produces the correct density at a given state, the distance to Mars center of mass is better. It would be good to make this user-configurable, but this can be done in a later step.

What would updating the unit tests entail? Are you reusing tests from the MCD library that have zkey=3? If so, then let's add an option now to make this settable in our atmosphere model. Then the tests can use zkey=3, but the default can go to zkey=1

Also, I see there are many changes to code files that are unrelated to this issue. Could you modify the PR so that the changes are only those that address #382 ? Thanks!

valeriof7 commented:
It sounds like a good plan! 

Regarding the PR, all the additional files come from the merge I did of the develop branch before the PR. Specifically, all the changes come from correcting the misspelled ancillary word. How is this possible?

DominicDirkx commented:
Great! Which one of you will make the changes?

For the branches, not sure what happened but you probably branched off of your local branch with changes rather than the develop branch. With a little luck this can be merged soon: https://github.com/tudat-team/tudatpy/pull/606 and there will be no more inconsistencies

nikemais commented:
I have been working on the next PR, reusing and refactoring this code (essentially incorporating the changes from this one). Tomorrow or early next week, I plan to create the draft, and then we can decide how to move forward (after #606 is merged, anyway).

DominicDirkx commented:
@nikemais So, shall we put this proposed modification of the zkey in the next PR that you're now working on?

nikemais commented:
Yes exactly. It would also supersede this PR, as the structure is different but with the same features. Ideally, we could also close this one if we are happy with the new structure (I want to stress that the work was not in vain and has been reused). However, it's nothing a merge cannot resolve, so merging this one into develop first is also an option.

valeriof7 commented:
I'm not following. Nicoló, could you please explain in detail what you're doing and how? Is it this #586? If so, please merge mine first and then modify.

Also, remember to use the board and move the issue to "In Progress." Otherwise, it is difficult to follow what is being worked on and by whom.

nikemais commented:
Yes, it's for #586. We can definitely merge this first.

In my PR, mcdAtmosphereModel remains similar and requires a MarsClimateDatabase object, which comprises the MCD exposure and caching logic. The idea is that MCD becomes an attribute (a new climateModel variable) of the planet Body object so that all the models that require it can access the same one, which gets updated only once. More info and a proper explanation will be in the PR.

DominicDirkx commented:
@valeriof7 Your changes to spelling of ancillary settings (and some other auto-formatting?) are leading to a lot of merge conflicts. Could you remove these changes from the branch, or resolve the merge conclifts in favor of the develop branch?

valeriof7 committed:
Merge remote-tracking branch 'upstream/develop' into feature/mcd

valeriof7 commented:
Easier than expected. Now the branch is ready to be merged before Nicoló's PR.

DominicDirkx commented:
Thanks!
